jax/interpreters/partial_eval.py

```diff
@@ -473,12 +466,8 @@ def tracers_to_jaxpr(in_tracers, out_tracers):
   core.skip_checks or core.check_jaxpr(jaxpr)
   return jaxpr, const_vals, env_vals
 
-
-def eqn_parents(eqn):
-  subjaxpr_tracers = [c for _, c in eqn.bound_subjaxprs]
-  return list(it.chain(eqn.invars,  *subjaxpr_tracers))
-
-def closure_convert_jaxpr(jaxpr):
+@cache()
```

Comment:
This is very tricky. The compilation caching keys depend on many pieces of the transformation state, including among other things, the pointer-value of the Jaxpr. Since in the previouscommit I have added a `convert_constvars_jaxpr` call in `process_call`, every time that one ran, it produces the same semantic Jaxpr but a different pointer value, which was breaking caching. 

I found this only because of `api_test.test_grad_of_jit_compilation_caching`. This is great! But I suspect that we had other caching bugs, e.g., in process_map, where `convert_constvars_jaxpr` was always present. 

The solution is to cache the constvars conversion.